docs/reference/commands.rst

```diff
@@ -41,16 +62,144 @@ upgrade phase. Refer to the output below for a description of all available opti
+      --verbose, -v         Increase logging verbosity in STDOUT.
+                            Multiple 'v's yield progressively more detail (up to 4).
+                            Note that by default the logfile will include standard logs
+                            from juju and websockets, as well as debug logs from all other
+                            modules. To also include the debug level logs from juju and
+                            websockets modules, use the maximum verbosity.
+      --quiet, -q           Disable output in STDOUT.
+      --auto-approve        Automatically approve and continue with each upgrade step without prompt.
+
+For upgrading **hypervisors**, in addition to the common options also found in
+**data-plane** upgrades, users can specify either **--machine** or **--az** to
+narrow the upgrade to a particular subset of nodes.
 
+.. terminal:: 
+    :input: cou upgrade hypervisors --help
+
+    Usage: cou upgrade hypervisors [options]
+
+    Upgrade machines with nova-compute and colocated services.
+    This is possible only if control-plane has been fully upgraded,
+    otherwise an error will be thrown.
+
+    Note that only principal applications colocated with nova-compute units
+    that support action-managed upgrades are within the scope of this command.
+    Other principal applications (e.g. ceph-osd) and subordinates
+    can be upgraded via the data-plane subcommand.
+
+    Options:
+      -h, --help            Show this help message and exit.
+      --model MODEL_NAME    Set the model to operate on.
+                            If not set, the currently active Juju model will be used.
+      --backup, --no-backup
+                            Include database backup step before cloud upgrade.
+                            Default to enabling database backup.
+      --force               Force the plan/upgrade of non-empty hypervisors.
+      --verbose, -v         Increase logging verbosity in STDOUT.
+                            Multiple 'v's yield progressively more detail (up to 4).
+                            Note that by default the logfile will include standard logs
+                            from juju and websockets, as well as debug logs from all other
+                            modules. To also include the debug level logs from juju and
+                            websockets modules, use the maximum verbosity.
+      --quiet, -q           Disable output in STDOUT.
+      --machine MACHINES, -m MACHINES
+                            Specify machine id(s) to upgrade.
+                            This option accepts a single machine id as well as a stringified
+                            comma-separated list of ids, and can be repeated multiple times.
+                            This option cannot be used together with [--availability-zone/--az].
+      --availability-zone AVAILABILITY_ZONES, --az AVAILABILITY_ZONES
```

Comment:
From what we talked with managed solutions, it seems that the concept of `availability-zone` can be considered also as OpenStack azs witch are different from `Juju`. I think if we add this information in the help message might be good for users.

Reply:
Good idea. Updated.